### ISSUE TITLE
travis-ci clang linux build is failing with usnic

### ISSUE THREAD

**Issue by j-xiong** [ofiwg/libfabric#3619]
All recent PR checkings are failing due to this error:

```
prov/usnic/src/usnic_direct/usd_dest.c:218:11: error: taking address of packed member 'uh_ip' of class or structure 'usd_udp_hdr' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
    ip = &dest->ds_dest.ds_udp.u_hdr.uh_ip;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
prov/usnic/src/usnic_direct/usd_dest.c:228:12: error: taking address of packed member 'uh_udp' of class or structure 'usd_udp_hdr' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
    udp = &dest->ds_dest.ds_udp.u_hdr.uh_udp;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
prov/usnic/src/usnic_direct/usd_dest.c:360:12: error: taking address of packed member 'uh_udp' of class or structure 'usd_udp_hdr' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
    udp = &dest->ds_dest.ds_udp.u_hdr.uh_udp;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
make[1]: *** [prov/usnic/src/usnic_direct/src_libfabric_la-usd_dest.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/home/travis/build/ofiwg/libfabric'
make: *** [all] Error 2
The command "if [[ "$TRAVIS_OS_NAME" == "linux" && "`basename $CC`" == "clang" ]]; then ./configure CFLAGS="-Werror $CFLAGS" $LIBFABRIC_CONFIGURE_ARGS --enable-debug && make -j2; fi" failed and exited with 2 during .
Your build has been stopped.
```

**Comment by shefty** [comment]
@bturrubiates @jsquyres -- I'm guessing travis updated clang versions or something that's showing this warning now.

**Comment by bturrubiates** [comment]
> @bturrubiates @jsquyres -- I'm guessing travis updated clang versions or something that's showing this warning now.

Huh, yeah I haven't seen this error before. I can check which version of Clang it's using and try to see if I can get a fix tomorrow.

**Comment by shefty** [comment]
I believe it's clang 4.0 or newer that causes this.

**Comment by jsquyres** [comment]
Should be fixed in #3627.
